include/seastar/core/sized-free.hh

```diff
@@ -0,0 +1,33 @@
+/*
+ * This file is open source software, licensed to you under the terms
+ * of the Apache License, Version 2.0 (the "License").  See the NOTICE file
+ * distributed with this work for additional information regarding copyright
+ * ownership.  You may not use this file except in compliance with the License.
+ *
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+// Copyright (C) 2025-present ScyllaDB
+
+#pragma once
+
+#include <cstddef>
+
+namespace seastar::memory {
+
+/// Deallocate memory with known size.
+///
+/// Similar to C++ sized deallocation, but for the C ::malloc() family instead
+/// of `operator new`. The size must be the same as the size used when allocating.
+void free(void* ptr, size_t size);
```

Comment:
Why does it need to become part of official API? Next patch can just call memory::do_free()